Duplicati/Library/Backend/Tardigrade/TardigradeBackend.cs

```diff
@@ -44,6 +45,26 @@ public class Tardigrade : IStreamingBackend
             { "Access grant", "Access grant" },
         };
 
+        [DllImport("kernel32.dll")]
+        protected static extern IntPtr LoadLibrary(string filename);
+
+        private static bool _libraryLoaded = false;
+        private static void InitStorjLibrary()
+        {
+            if (_libraryLoaded)
+                return;
+
+            if(System.Environment.Is64BitProcess)
+            {
+               var res = LoadLibrary("win-x64/storj_uplink.dll");
+            }
+            else
+            {
+                var res = LoadLibrary("win-x86/storj_uplink.dll");
+            }
+            _libraryLoaded = true;
```

Comment:
Instead of keeping track of `_libraryLoaded`, what are your thoughts about doing the above in a static constructor?

Reply:
I wanted to be sure that the DLL gets loaded before everythin else. But I assume this now leads to the situation that the DLL gets loaded whenever the BackendLoader collects the available backends, right? Didn't think of that.

Reply:
Good point.  We should probably avoid loading the library if it's not necessary.

Any other changes you want to make before I merge?

Reply:
Resolved this, too. I'm fine for merging! Thank you! @warwickmm

Reply:
Thanks!  One last question.  I noticed when I build the solution that a file `libstorj_uplink.so` appears in the output directory.  How is this file appearing without the treatment being applied in this pull request?

Reply:
You should also see a libstorj_uplink.dylib. The idea is to place all binaries for the different platforms in the output dir and include it in the installer. But I've also adjusted the build scripts for the platforms to remove all files that are not necessary there. So the *.so and *.dylib would be removed on Windows, the dylib and dll would be removed on linux and so on. You can see this in the other files I've change with this PR.

Reply:
Hmmm...I don't see `libstorj_uplink.dylib`.  This is after a rebuild of the solution:
```
$ ls -1 Duplicati/GUI/Duplicati.GUI.TrayIcon/bin/Debug/*.{so,dylib}
Duplicati/GUI/Duplicati.GUI.TrayIcon/bin/Debug/libstorj_uplink.so
Duplicati/GUI/Duplicati.GUI.TrayIcon/bin/Debug/storj_uplink.dylib
Duplicati/GUI/Duplicati.GUI.TrayIcon/bin/Debug/storj_uplink.so
```
How does `libstorj_uplink.so` appear in the output directory without the explicit `CopyToOutputDirectory` entries in the `.csproj` file that the other files need?

Reply:
There's a "storj_uplink.dylib" - that's ok.

Are you building duplicati on Linux?

Reply:
Yes, this is on Linux.

Reply:
So then the file comes from the "normal" build on linux. You won't have the storj_uplink.dll then.
The installer is build on windows, as I've understood? there you would habe the correct files then - the linux-so, the mac-dyli and the Window-dll.

I could change storj_uplink.so to libstorj_uplink.so so we only have one linux-binary in all cases.

Reply:
It looks like `libstorj_uplink.so` was added to the project as a link to the nuget package folder, while `storj_uplink.dylib` and `storj_uplink.so` were added separately.  Is there a reason for the difference?

![image](https://user-images.githubusercontent.com/17345532/86139690-e6ee4800-baa4-11ea-8eeb-f1da7bc55eba.png)

Reply:
> The installer is build on windows, as I've understood?

I don't build the installers (@kenkendk creates the releases), but my impression is that it's built on either Mac or Linux via the `build-release.sh` script.

Reply:
I've just commited a change. You should now only see one file.

Yes, the uplink.net deploys the binaries for the system you build on. But for the installer, we need all - so I've added them as assets to the Tardigrad-Backend-Project so they always get deployed to the output-dir. the installer-script should remove it from the platform-installers afterwards.

Reply:
Well, now it should work on all platforms as all three binaries (4 if you count win-x64 and win-x86 seperately) get deployed as assets. We should give it a try.

Reply:
> Yes, the uplink.net deploys the binaries for the system you build on. But for the installer, we need all - so I've added them as assets to the Tardigrad-Backend-Project so they always get deployed to the output-dir. the installer-script should remove it from the platform-installers afterwards.

Ah, ok.  That makes sense.  Thanks.

Reply:
One for Linux, one for Mac. Plus there should be win-DLLs in subfolders. storj_uplink.dll.dll

Reply:
Yup, I see them.  Looks good, I'll merge.